src/core/lib/iomgr/socket_utils_posix.cc

```diff
@@ -21,15 +21,104 @@
 #include "src/core/lib/iomgr/port.h"
 
 #ifdef GRPC_POSIX_SOCKETUTILS
-
 #include <fcntl.h>
 #include <sys/socket.h>
 #include <unistd.h>
 
+#include <grpc/impl/codegen/grpc_types.h>
 #include <grpc/support/log.h>
 
 #include "src/core/lib/iomgr/sockaddr.h"
 #include "src/core/lib/iomgr/socket_utils_posix.h"
+#endif
+
+#ifdef GRPC_POSIX_SOCKET_TCP
+
+#include "src/core/lib/event_engine/channel_args_endpoint_config.h"
+#include "src/core/lib/iomgr/socket_utils_posix.h"
+
+using ::grpc_event_engine::experimental::EndpointConfig;
+
+#define MAX_CHUNK_SIZE (32 * 1024 * 1024)
+
+using ::grpc_core::PosixTcpOptions;
+
+namespace {
+
+int Clamp(int default_value, int min_value, int max_value, int actual_value) {
+  if (actual_value < min_value || actual_value > max_value) {
+    return default_value;
+  }
+  return actual_value;
+}
+
+int GetConfigValue(const EndpointConfig& config, absl::string_view key,
```

Comment:
Adding helpers on the `grpc_core::ChannelArgs` type to do this was really helpful... maybe we could do the same here. Concretely, add `optional<int> GetInt(string_view name)` to `EndpointConfig` and write usages like `Clamp(config.GetInt(GRPC_ARG_TCP_READ_CHUNK_SIZE).value_or(PosixTcpOptions::kDefaultReadChunkSize), 1, PosixTcpOptions::kMaxChunkSize)`

Reply:
Added a GetInt, GetString and a GetPointer method to EndpointConfig interface.